src/test/java/com/nike/cerberus/client/DefaultCerberusClientFactoryTest.java

```diff
@@ -35,4 +45,69 @@ public void test_that_getClient_adds_client_version_as_a_default_header() {
                 result.getDefaultHeaders().get(ClientVersion.CERBERUS_CLIENT_HEADER));
     }
 
+    @Test
+    public void test_that_getClient_adds_client_version_as_a_default_header_and_returns_CerberusClientFactory() {
+        String region = "us-west-2";
+        String url = "url";
+        SSLSocketFactory sslSocketFactory = new SSLSocketFactory() {
```

Comment:
I'd look into using Mockito et.al. to mock the socket factory and trust manager in this test. It would be a cleaner read.

Reply:
i looked into it , but we are not making any call to one of the methods in these (socket factory & Trust manager ) methods , mocking doesn't make sense to me , what do you think ?

Reply:
John, please check the below commit and let me know how it looks